### DISCUSSION TITLE
Exposing Edge Function Metrics in the Prometheus-compatible Metrics Endpoint

### DISCUSSION THREAD

**Discussion by brettimus** [supabase/supabase#17269] [Feature Requests]
Hello! So, [the Prometheus-compatible metrics endpoint](https://supabase.com/docs/guides/platform/metrics) is cool, and really useful. 

One thing that would be even _cooler_ is if that endpoint exposed metrics for edge functions as well. (I'm drawing inspiration from [this Reddit post](https://www.reddit.com/r/Supabase/comments/14d22p4/how_do_you_improve_observability_on_supabase).)

Specifically, if the metrics endpoint exposed series that could help calculate:

- rate of invocations
- ratio of errors
- latency

I get the sense that exposing the request and error counts would be straightforward, though I know latency is a little tricky due to choosing the correct histogram buckets in Prometheus (or VictoriaMetrics, which I think you all are using at Supabase).

I'm not sure if this is already in the works, but if not, I'd also like to use this thread to suggest naming conventions for the series and labels, which would be similar to how we record function-level metrics in [an open-source project](https://github.com/autometrics-dev) I work on. Here's an example of a counter for function invocations that can track success/error:

```
# COUNTER of successful invocations
function_calls_total{
  function="<name-of-edge-function>",
  module="<supabase-project-id>",
  result="ok"
} 2

# COUNTER of erroneous invocations
function_calls_total{
  function="<name-of-edge-function>",
  module="<supabase-project-id>",
  result="error"
} 1
```

Excited to hear back, and keep up the awesome work! ⚡

**Comment by emul-docler** [comment]
Unfortunately, they don’t even bother to respond, even though in its current form the system is not really monitorable. :(

@brettimus Did you manage to solve this somehow?
